### ISSUE TITLE
Brotli compression support in ddev-webserver

### ISSUE THREAD

**Issue by acalvino4** [drud/ddev#4055]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

As far as I can tell, ddev serves all content with gzip. This has a small, but significant performance hit compared to brotli.

### Describe the solution you'd like

Since it now has [96.3% browser support](https://caniuse.com/?search=brotli), it should be perfectly save for developers to use (I imagine it's actually at 100% support in developer browsers). This will bring ddev up to date with the modern standard for web server compression.

Additionally, many of us use Brotli in production now, so this will enable greater parity between our environments.

Personally, I think brotli should be the default (possibly with an option to stick with gzip), and don't know of any reason why it shouldn't be. However, if it was a simple config option I wouldn't mind that either.

**Comment by rfay** [comment]
Hi @acalvino4 you should be able to configure this yourself very easily with a .ddev/nginx-full/nginx-site.conf that changes the compression config, or with a .ddev/web-build/Dockerfile.brotly that changes the base nginx config.

Why don't you try that? 

If you like it, it can be a `ddev-get` add-on recipe, and possibly make it into ddev later.

But remember that compared to docker itself, the cost of compression in serving a web page is completely trivial. And remember that ddev is a development environment, so you're not serving many pages anyway. 

If you need help figuring out how to configure in ddev, please provide instructions and links about how brotli is to be configured.

**Comment by acalvino4** [comment]
Yes, I agree that the speed is trivial, I think this is more of a standards issue - i.e. brotli is just the new norm since if you can't get 15-40% better (and faster) compression across the board for all users of your product with a simple config change, why wouldn't you?

I think the bigger issue is that I need a conditional in my build process to not compress to brotli on local, and while the conditional is trivial, good programming practice demands that we at least _try_ to eliminate complexity where we can...

Anyway, I did try extending the nginx config, but I don't think the nginx version that ddev uses has the brotli module built in. (On ubuntu, nginx 1.20 is the first to have it in the default install). When I use a directive like `brotli_static on` for my build directory, I get an error that the directive is unknown.

Bottom line, since ddev is about improving developer experience, and brotli is the new norm for compression, I think it should be built in, though I understand it takes effort and time and am happy to use a workaround config in the meantime. On that front though, I think there still may be changes required in the underlying nginx version, though if there is a way of configuring that I'd be happy to be pointed in the right direction.

**Comment by rfay** [comment]
Current ddev-websever has  nginx/1.22.0

Seems nginx only has built-in support  for brotli with nginx plus. https://www.nginx.com/products/nginx/modules/brotli/

The non-built-in technique requires lots of hoop-jumping it looks like: https://dev.to/requestmetrics/using-brotli-compression-in-nginx-22ge, https://github.com/google/ngx_brotli#installation

You can do those hoops in a Dockerfile if that's important. 

I don't see adding this to default DDEV in this state of its life, if it's not even arrived in nginx yet and there are no package-installable packages.

**Comment by acalvino4** [comment]
Interesting....I've been serving brotli from nginx 1.22 (I think even on 1.20) on Ubuntu for several months now. I guess that was just a choice by the debian package people to compile/bundle that module in by default. 

Maybe one day I'll have time and take a stab at it for whatever container ddev is using.

**Comment by rfay** [comment]
ddev-webserver is Debian bullseye. Maybe Ubuntu 22.04 has an nginx package with the extra module? I'd think that would show in *some* instructions. You can easily test this by manually running `docker run -it --rm ubuntu:22.04 bash` and manually installing nginx and seeing if the module is there. Try it with 20.04 and debian:bullseye as well. 

Note that ddev uses the official nginx upstream packages, not the ones provided by the debian apt repo, so that could be a difference also.

**Comment by rfay** [comment]
Happy to have a PR when it becomes possible. I confirm that the nginx shipped with Ubuntu 22.04 does not have brotli in it, and it doesn't appear that it is installable with packages. 

Debian Bullseye comes with nginx 1.18, and it also does not have brotli in it. 

I'm going to close this for now, but happy to have a PR when brotli is more widely available.

**Comment by acalvino4** [comment]
Figured out how I have brotli enabled despite using the community edition - I've been using the `ppa:ondrej/nginx` repository, which has it compiled it:
https://stackoverflow.com/a/67631812/2410203

**Comment by rfay** [comment]
You're using that in the ddev-webserver? Of course the ondrej PPA is intended for Ubuntu, not Debian, and by using ondrej/nginx you'll change the nginx package right?  (Since DDEV uses deb.sury.org's Debian packages (From ondrej), there ought to be a way to just pick this up from there.)

Can you please give your full approach here for those that might want it?

**Comment by acalvino4** [comment]
No - I just meant I've been using it in my servers, and was explaining my previous comment that said "I've been serving brotli from nginx 1.22"

**Comment by rfay** [comment]
I don't think an analog of ondrej's Ubuntu nginx PPA is available for Debian, but if there were that would be interesting.

We'll see if this feature request gets some traction, and please check in if you find a reputable Debian nginx package that has brotli.
